app/controllers/WhatIsYourNameController.scala

```diff
@@ -77,25 +76,9 @@ class WhatIsYourNameController @Inject() (
             formWithErrors => render(mode, regime, formWithErrors).map(BadRequest(_)),
             value =>
               for {
-                updatedAnswers <- Future.fromTry(request.userAnswers.set(WhatIsYourNamePage, value))
-                _              <- sessionRepository.set(updatedAnswers)
+                updatedAnswers <- Future.fromTry(request.userAnswers.setOrCleanup(WhatIsYourNamePage, value, true))
+                _ = sessionRepository.set(updatedAnswers)
               } yield Redirect(navigator.nextPage(WhatIsYourNamePage, mode, regime, updatedAnswers))
```

Comment:
I think this is a Future and call and should not have  _  = sessionRepository.set(updatedAnswers)

Reply:
I agree